### PR TITLE
Expose metrics and add Prometheus ServiceMonitor

### PR DESCRIPTION
**Pull request by modularTaco**
OAuth2-Proxy now supports metrics, so I think it would make sense to expose those metrics and scrape them with a prometheus-operator ServiceMonitor.

### CODE REVIEW

**Review by pierluigilenoci — DISMISSED** [review]
LGTM

**Comment by pierluigilenoci** [comment]
@modularTaco could you please fix the error? More info [here](https://github.com/oauth2-proxy/manifests/pull/46/checks?check_run_id=3007945738)
```sh
Error: [trailing-spaces] trailing spaces
```

**Comment by modularTaco** [comment]
> @modularTaco could you please fix the error? More info [here](https://github.com/oauth2-proxy/manifests/pull/46/checks?check_run_id=3007945738)
> 
> ```shell
> Error: [trailing-spaces] trailing spaces
> ```

trailing space has been removed in the latest commit

**Comment by pierluigilenoci** [comment]
@modularTaco could you please bump the chart again?

**Comment by modularTaco** [comment]
> @modularTaco could you please bump the chart again?

@pierluigilenoci Which version is desired? I thought `4.1.0` would be correct as `4.0.5` is the current version and this is a minor change

**Comment by pierluigilenoci** [comment]
> @pierluigilenoci Which version is desired? I thought `4.1.0` would be correct as `4.0.5` is the current version and this is a minor change

I got confused, a rebase was enough. Then version 4.1.0 is fine. At least I think so. 
I would also like the opinion of another maintainer before doing the merge though.

**Comment by modularTaco** [comment]
> > @pierluigilenoci Which version is desired? I thought `4.1.0` would be correct as `4.0.5` is the current version and this is a minor change
> 
> I got confused, a rebase was enough. Then version 4.1.0 is fine. At least I think so.
> I would also like the opinion of another maintainer before doing the merge though.

OK thank you :+1:

**Review by desaintmartin — COMMENTED** [review]
Thanks! Here is a comment.

File: helm/oauth2-proxy/README.md
Comment (modularTaco) [reply helm/oauth2-proxy/README.md]: I've changed it now to `metrics` instead of `prometheus`

**Review by pierluigilenoci — APPROVED** [review]

**Review by desaintmartin — APPROVED** [review]

**Comment by desaintmartin** [comment]
Conflicts :(

**Comment by modularTaco** [comment]
> Conflicts :(

Should be resolved now

**Comment by desaintmartin** [comment]
Thanks!

**Pull request merged**
